chain/jsonrpc/src/api/validator.rs

```diff
@@ -49,3 +55,73 @@ impl RpcFrom<GetValidatorInfoError> for RpcValidatorError {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::api::RpcRequest;
+    use near_jsonrpc_primitives::types::validator::RpcValidatorRequest;
+    use near_primitives::hash::CryptoHash;
+    use near_primitives::types::{BlockId, EpochId, EpochReference};
+
+    #[test]
+    fn test_serialize_validators_params_as_vec() {
+        let block_hash = CryptoHash::new();
+        let block_hash_string = CryptoHash::new().to_string();
+        let params = serde_json::json!([block_hash_string]);
+        let result = RpcValidatorRequest::parse(params);
+        assert!(result.is_ok());
+        let result_unwrap = result.unwrap();
+        let res_serialized = format!("{result_unwrap:?}");
+        let expected = RpcValidatorRequest {
+            epoch_reference: EpochReference::BlockId(BlockId::Hash(block_hash)),
+        };
+        let expected_serialized = format!("{expected:?}");
+        assert_eq!(res_serialized, expected_serialized);
+    }
+
+    #[test]
+    fn test_serialize_validators_params_as_object_input_block_hash() {
+        let block_hash = CryptoHash::new();
+        let block_hash_string = CryptoHash::new().to_string();
+        let params = serde_json::json!({"block_id": block_hash_string});
+        let result = RpcValidatorRequest::parse(params);
+        assert!(result.is_ok());
+        let result_unwrap = result.unwrap();
+        let res_serialized = format!("{result_unwrap:?}");
+        let expected = RpcValidatorRequest {
+            epoch_reference: EpochReference::BlockId(BlockId::Hash(block_hash)),
+        };
+        let expected_serialized = format!("{expected:?}");
+        assert_eq!(res_serialized, expected_serialized);
+    }
+
+    #[test]
+    fn test_serialize_validators_params_as_object_input_block_height() {
+        let block_height: u64 = 12345;
+        let params = serde_json::json!({"block_id": block_height});
```

Comment:
Very nice, it works the same as we have in blocks API (the ability of parsing both height and hash under the same name)
https://docs.near.org/api/rpc/block-chunk#block-details

Could you please update docs repo as well? Mentioing that we have a new format now
Be careful and don't merge it until we release 1.37

Reply:
Hey Olga, what is the docs repo you are referring to?
And when you say don't merge it until we release 1.37, do you mean the change to the docs repo for the api change?

Reply:
https://github.com/near/docs

> do you mean the change to the docs repo

yes ofk
It's because the changes in nearcore will be live only after the release of 1.37, while the docs will be updated right after the merge
My PR is also waiting https://github.com/near/docs/pull/1564

Reply:
Got it! Will open a PR in docs tmr.